src/main/kotlin/com/ajaxproject/telegrambot/bot/service/UserSessionService.kt

```diff
@@ -0,0 +1,15 @@
+package com.ajaxproject.telegrambot.bot.service
+
+import com.ajaxproject.telegrambot.bot.model.UserSession
+import org.springframework.stereotype.Component
+
+@Component
+class UserSessionService {
+    val userSessionMap: MutableMap<Long, UserSession> = mutableMapOf()
+    fun getSession(chatId: Long): UserSession {
+        return userSessionMap.getOrDefault(chatId, UserSession(chatId = chatId))
+    }
+    fun saveSession(chatId: Long, session: UserSession): UserSession {
+        return userSessionMap.put(chatId, session)!!
```

Comment:
It's wierd logic.

You save a session here and return the previous session(look at javadoc of `MutableMap#put` function).

Seems, you should return the actual session:
```suggestion
        userSessionMap[chatId] = session
        return session
```

It really looks like a bug